src/flickr-api-object.js

```diff
@@ -22,9 +22,9 @@ module.exports = (function() {
       }, 1);
       return;
     }
-
-    var method_name = methods[method_idx],
-        mdir = "data/flickr/methods",
+    
+    var mdir = (flickrOptions.data_path || "./data") + "/flickr/methods",
```

Comment:
given that we can now not trust that the path exists, this will need a requirement of the `mkdir` package (or `fs-extra`) so that we ensure the dir will exist before we start writing files to it.